backend/src/server/app.ts

```diff
@@ -145,6 +156,71 @@ export const main = async ({
       })
     });
 
+    // Health check - returns 200 only if doing useful work (running migrations or ready)
+    // Returns 503 if waiting for another container to finish migrations
+    server.get("/api/health", async (_, reply) => {
```

Comment:
if it gets 503, i think it will try to kill the container and then it will spint up again?